compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java

```diff
@@ -589,6 +590,21 @@ private void addToCompiledSymbols(Set<Symbol> compiledSymbols, Scope.ScopeEntry
         addToCompiledSymbols(compiledSymbols, scopeEntry.next, cursorPos, name, symbolEnv, states, compUnitName);
     }
 
+    private boolean checkAndUpdateModuleSymbols(Set<Symbol> compiledSymbols, Symbol evaluatingSymbol, BSymbol symbol) {
+        boolean isContaining = compiledSymbols.contains(evaluatingSymbol);
+
+        if (evaluatingSymbol.kind() != MODULE) {
+            return isContaining;
+        }
+
+        if (((BPackageSymbol) symbol).importPrefix != null) {
+            compiledSymbols.remove(evaluatingSymbol);
+            return false;
+        }
+
+        return isContaining;
+    }
+
```

Comment:
With this logic, we are removing the already added symbol. Why don't we have this logic when appending to the compiled-symbols (to ignore the new symbol)?

Reply:
Had an offline chat with @sanjana. It seems to be we have two symbols and prefix information is carried by the second module symbol. Let's have a comment in here and go ahead with this approache

Reply:
Added in https://github.com/ballerina-platform/ballerina-lang/pull/38093/commits/027b54d297c9bb3ae6dfc669a303f33ac95404f2.